transaction.go

```diff
@@ -57,7 +57,7 @@ func (t *Tracer) StartTransactionOptions(name, transactionType string, opts Tran
 	// transaction.
 	instrumentationConfig := t.instrumentationConfig()
 	tx.recording = instrumentationConfig.recording
-	if !tx.recording {
+	if !t.Recording() {
```

Comment:
```suggestion
	if !tx.recording || !t.Active() {
```

`t.Recording()` internally calls `t.instrumentationConfig()` to fetch a config snapshot, which we've already done above. Using a single config snapshot for all transaction behaviour makes it easier to reason about.